Trench Warfare/map.xml

```diff
@@ -0,0 +1,355 @@
+</filters>
+<regions>
+    <negative id="playable-inverse">
+        <union id="playable">
+            <rectangle id="main-area" min="59,-10" max="-60,11"/>
+            <union id="bases">
+                <union id="red-base-area">
+                    <rectangle id="red-base" min="60,-17" max="76,18"/>
+                    <rectangle id="red-side-right" min="59,-11" max="44,-17"/>
+                    <rectangle id="red-side-left" min="59,12" max="44,18"/>
+                </union>
+                <union id="blue-base-area">
+                    <rectangle id="blue-base" min="-61,18" max="-77,-17"/>
+                    <rectangle id="blue-side-left" min="-60,12" max="-45,18"/>
+                    <rectangle id="blue-side-right" min="-60,-11" max="-45,-17"/>
+                </union>
+            </union>
+        </union>
+    </negative>
+    <union id="cannons">
+        <cuboid id="red-cannon1" min="69,74,6" max="65,78,7"/>
+        <cuboid id="red-cannon2" min="72,74,-0" max="63,77,2"/>
+        <cuboid id="red-cannon3" min="69,74,-4" max="65,78,-5"/>
+        <cuboid id="blue-cannon1" min="-69,74,-5" max="-65,78,-4"/>
+        <cuboid id="blue-cannon2" min="-72,74,2" max="-63,77,0"/>
+        <cuboid id="blue-cannon3" min="-69,74,6" max="-65,78,7"/>
+    </union>
+    <apply enter="only-red" region="red-base-area" message="Only red can enter this area!"/>
+    <apply enter="only-blue" region="blue-base-area" message="Only blue can enter this area!"/>
+    <apply block="only-tnt" region="cannons"/>
+    <apply block="no-tnt" region="main-area" message="You may only place TNT inside cannons!"/>
+    <apply block="never" region="bases" message="You cannot modify the base!"/>
+    <apply block="break-restrictions" region="main-area" message="You may not break this block!"/>
+    <apply block="never" region="playable-inverse"/>
+</regions>
+<control-points incremental="true" show-progress="true" required="false">
+    <control-point name="East Trench" capture-time="20s">
+        <capture><cuboid min="22,62,-7" max="26,69,9"/></capture>
+        <progress><cuboid min="21,67,-7" max="27,68,9"/></progress>
+        <captured><cuboid min="22,62,-7" max="26,69,9"/></captured>
+    </control-point>
+    <control-point name="Middle Trench" capture-time="30s" points="1.25">
+        <capture><cuboid min="-2,62,-7" max="2,69,9"/></capture>
+        <progress><cuboid min="-3,66,-7" max="3,67,9"/></progress>
+        <captured><cuboid min="-2,62,-7" max="2,69,9"/></captured>
+    </control-point>
+    <control-point name="West Trench" capture-time="20s">
+        <capture><cuboid min="-26,62,-7" max="-22,69,9"/></capture>
+        <progress><cuboid min="-27,67,-7" max="-21,68,9"/></progress>
+        <captured><cuboid min="-26,62,-7" max="-22,69,9"/></captured>
+    </control-point>
+</control-points>
+<projectiles>
+    <projectile
+        id="gun"
+        name="gun"
+        projectile="TNTPrimed"
+        velocity="0.5"
+        damage="20"
```

Comment:
`0.5` velocity seems good, almost just "throwing" out a TNT, and grenade launcher don't shoot very far I think

Not sure how much damage it should do though. I think it should maybe do less than what a normal TNT do.